kv/kvserver: TestSystemZoneConfigs failed

kv/kvserver.TestSystemZoneConfigs [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8965166?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8965166?buildTab=artifacts#/) on master @ [1b162d1b274eec7b307fbbfca7294460bfdef025](https://github.com/cockroachdb/cockroach/commits/1b162d1b274eec7b307fbbfca7294460bfdef025):


```
=== RUN   TestSystemZoneConfigs
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9ded2e1b6e3cd0145900033ce06c11b0/logTestSystemZoneConfigs2842108216
    test_log_scope.go:79: use -show-logs to present logs inline
    client_replica_test.go:2768: condition failed to evaluate within 45s: got 288 voters, want 290; details: map[1:r1:/{Min-System/NodeLiveness} [(n1,s1):1, (n4,s4):2, (n6,s6):3, (n2,s2):4, (n7,s7):5, next=6, gen=8] 2:r2:/System/NodeLiveness{-Max} [(n1,s1):1, (n3,s3):2, (n4,s4):3, (n6,s6):4, (n7,s7):5, next=6, gen=8] 3:r3:/System/{NodeLivenessMax-tsd} [(n1,s1):1, (n6,s6):2, (n7,s7):3, (n3,s3):4, (n5,s5):5, next=6, gen=8] 4:r4:/System{/tsd-tse} [(n3,s3):4, (n6,s6):2, (n2,s2):3, next=5, gen=8] 5:r5:/{Systemtse-Table/0} [(n1,s1):1, (n2,s2):2, (n5,s5):3, (n3,s3):4, (n6,s6):5, next=6, gen=8] 6:r6:/Table/{0-3} [(n1,s1):1, (n7,s7):2, (n5,s5):3, (n6,s6):4, (n2,s2):5, next=6, gen=8] 7:r7:/Table/{3-4} [(n1,s1):1, (n5,s5):2, (n2,s2):3, (n7,s7):4, (n4,s4):5, next=6, gen=8] 8:r8:/Table/{4-5} [(n1,s1):1, (n5,s5):2, (n3,s3):3, (n4,s4):4, (n2,s2):5, next=6, gen=8] 9:r9:/Table/{5-6} [(n2,s2):6, (n3,s3):2, (n5,s5):3, (n4,s4):4, (n6,s6):5, next=7, gen=12] 10:r10:/Table/{6-7} [(n1,s1):1, (n5,s5):2, (n7,s7):3, (n4,s4):4, (n2,s2):5, next=6, gen=8] 11:r11:/Table/{7-8} [(n1,s1):1, (n6,s6):2, (n4,s4):3, (n3,s3):4, (n2,s2):5, next=6, gen=8] 12:r12:/Table/{8-11} [(n1,s1):1, (n5,s5):2, (n2,s2):3, (n3,s3):4, (n4,s4):5, next=6, gen=8] 13:r13:/Table/1{1-2} [(n1,s1):1, (n6,s6):2, (n3,s3):3, (n4,s4):4, (n7,s7):5, next=6, gen=8] 14:r14:/Table/1{2-3} [(n1,s1):1, (n7,s7):2, (n4,s4):3, (n2,s2):4, (n3,s3):5, next=6, gen=8] 15:r15:/Table/1{3-4} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n3,s3):4, (n7,s7):5, next=6, gen=8] 16:r16:/Table/1{4-5} [(n5,s5):6, (n3,s3):2, (n4,s4):3, (n7,s7):4, (n2,s2):5, next=7, gen=12] 17:r17:/Table/1{5-6} [(n1,s1):1, (n5,s5):2, (n3,s3):3, (n6,s6):4, (n7,s7):5, next=6, gen=8] 18:r18:/Table/1{6-7} [(n3,s3):6, (n4,s4):2, (n5,s5):3, (n6,s6):4, (n7,s7):5, next=7, gen=12] 19:r19:/Table/1{7-8} [(n1,s1):1, (n4,s4):2, (n6,s6):3, (n2,s2):4, (n5,s5):5, next=6, gen=8] 20:r20:/Table/1{8-9} [(n1,s1):1, (n7,s7):2, (n2,s2):3, (n5,s5):4, (n6,s6):5, next=6, gen=8] 21:r21:/Table/{19-20} [(n1,s1):1, (n7,s7):2, (n2,s2):3, (n6,s6):4, (n5,s5):5, next=6, gen=8] 22:r22:/Table/2{0-1} [(n1,s1):1, (n7,s7):2, (n6,s6):3, (n4,s4):4, (n5,s5):5, next=6, gen=8] 23:r23:/Table/2{1-2} [(n1,s1):1, (n7,s7):2, (n6,s6):3, (n3,s3):4, (n4,s4):5, next=6, gen=8] 24:r24:/Table/2{2-3} [(n1,s1):1, (n7,s7):2, (n6,s6):3, (n5,s5):4, (n3,s3):5, next=6, gen=8] 25:r25:/Table/2{3-4} [(n4,s4):6, (n2,s2):2, (n3,s3):3, (n7,s7):4, (n6,s6):5, next=7, gen=12] 26:r26:/Table/2{4-5} [(n5,s5):6, (n2,s2):2, (n4,s4):3, (n3,s3):4, (n7,s7):5, next=7, gen=12] 27:r27:/Table/2{5-6} [(n2,s2):6, (n4,s4):2, (n7,s7):3, (n5,s5):4, (n3,s3):5, next=7, gen=12] 28:r28:/Table/2{6-7} [(n1,s1):1, (n4,s4):2, (n2,s2):3, (n5,s5):4, (n3,s3):5, next=6, gen=8] 29:r29:/Table/2{7-8} [(n4,s4):6, (n3,s3):2, (n2,s2):3, (n7,s7):4, (n6,s6):5, next=7, gen=12] 30:r30:/Table/2{8-9} [(n1,s1):1, (n4,s4):2, (n5,s5):3, (n2,s2):4, (n7,s7):5, next=6, gen=8] 31:r31:/{Table/29-NamespaceTable/30} [(n1,s1):1, (n7,s7):2, (n4,s4):3, (n3,s3):4, (n6,s6):5, next=6, gen=8] 32:r32:/NamespaceTable/{30-Max} [(n1,s1):1, (n5,s5):2, (n4,s4):3, (n2,s2):4, (n6,s6):5, next=6, gen=8] 33:r33:/{NamespaceTable/Max-Table/32} [(n1,s1):1, (n5,s5):2, (n3,s3):3, (n7,s7):4, (n4,s4):5, next=6, gen=8] 34:r34:/Table/3{2-3} [(n1,s1):1, (n6,s6):2, (n7,s7):3, (n3,s3):4, (n2,s2):5, next=6, gen=8] 35:r35:/Table/3{3-4} [(n6,s6):6, (n4,s4):2, (n3,s3):3, (n5,s5):4, (n7,s7):5, next=7, gen=12] 36:r36:/Table/3{4-5} [(n1,s1):1, (n3,s3):2, (n5,s5):3, (n4,s4):4, (n2,s2):5, next=6, gen=8] 37:r37:/Table/3{5-6} [(n1,s1):1, (n3,s3):2, (n7,s7):3, (n2,s2):4, (n4,s4):5, next=6, gen=8] 38:r38:/Table/3{6-7} [(n1,s1):1, (n6,s6):2, (n5,s5):3, (n2,s2):4, (n3,s3):5, next=6, gen=8] 39:r39:/Table/3{7-8} [(n1,s1):1, (n3,s3):2, (n7,s7):3, (n5,s5):4, (n6,s6):5, next=6, gen=8] 40:r40:/Table/3{8-9} [(n6,s6):6, (n4,s4):2, (n2,s2):3, (n7,s7):4, (n5,s5):5, next=7, gen=12] 41:r41:/Table/{39-40} [(n1,s1):1, (n6,s6):2, (n7,s7):3, (n5,s5):4, (n4,s4):5, next=6, gen=8] 42:r42:/Table/4{0-1} [(n1,s1):1, (n6,s6):2, (n7,s7):3, (n5,s5):4, (n3,s3):5, next=6, gen=8] 43:r43:/Table/4{1-2} [(n3,s3):6, (n2,s2):2, (n7,s7):3, (n4,s4):4, (n6,s6):5, next=7, gen=12] 44:r44:/Table/4{2-3} [(n1,s1):1, (n6,s6):2, (n3,s3):3, (n2,s2):4, (n5,s5):5, next=6, gen=8] 45:r45:/Table/4{3-4} [(n1,s1):1, (n5,s5):2, (n3,s3):3, (n6,s6):4, (n4,s4):5, next=6, gen=8] 46:r46:/Table/4{4-5} [(n2,s2):6, (n3,s3):2, (n6,s6):3, (n4,s4):4, (n5,s5):5, next=7, gen=12] 47:r47:/Table/4{5-6} [(n1,s1):1, (n2,s2):2, (n6,s6):3, (n7,s7):4, (n5,s5):5, next=6, gen=8] 48:r48:/Table/4{6-7} [(n1,s1):1, (n5,s5):2, (n2,s2):3, (n6,s6):4, (n4,s4):5, next=6, gen=8] 49:r49:/Table/4{7-8} [(n1,s1):1, (n5,s5):2, (n2,s2):3, (n4,s4):4, (n3,s3):5, next=6, gen=8] 50:r50:/Table/{48-50} [(n1,s1):1, (n7,s7):2, (n6,s6):3, (n5,s5):4, (n2,s2):5, next=6, gen=8] 51:r51:/Table/5{0-1} [(n4,s4):6, (n2,s2):2, (n6,s6):3, (n3,s3):4, (n7,s7):5, next=7, gen=12] 52:r52:/Table/5{1-2} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n6,s6):4, (n7,s7):5, next=6, gen=8] 53:r53:/Table/5{2-3} [(n1,s1):1, (n6,s6):2, (n5,s5):3, (n7,s7):4, (n3,s3):5, next=6, gen=8] 54:r54:/Table/5{3-4} [(n6,s6):6, (n2,s2):2, (n7,s7):3, (n5,s5):4, (n4,s4):5, next=7, gen=12] 55:r55:/Table/5{4-5} [(n7,s7):6, (n4,s4):2, (n5,s5):3, (n2,s2):4, (n3,s3):5, next=7, gen=12] 56:r56:/Table/5{5-6} [(n1,s1):1, (n3,s3):2, (n4,s4):3, (n5,s5):4, (n7,s7):5, next=6, gen=8] 57:r57:/Table/5{6-7} [(n1,s1):1, (n3,s3):2, (n2,s2):3, (n4,s4):4, (n6,s6):5, next=6, gen=8] 58:r58:/{Table/57-Max} [(n1,s1):1, (n3,s3):2, (n6,s6):3, (n2,s2):4, (n5,s5):5, next=6, gen=8]]
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9ded2e1b6e3cd0145900033ce06c11b0/logTestSystemZoneConfigs2842108216
--- FAIL: TestSystemZoneConfigs (56.08s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSystemZoneConfigs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25127

kv/kvserver.TestSystemZoneConfigs [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9170520?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9170520?buildTab=artifacts#/) on master @ [1e9899da8cb250a4e560b280beb2b0805ee75a78](https://github.com/cockroachdb/cockroach/commits/1e9899da8cb250a4e560b280beb2b0805ee75a78):


```
=== RUN   TestSystemZoneConfigs
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestSystemZoneConfigs589320229
    test_log_scope.go:79: use -show-logs to present logs inline
    client_replica_test.go:2777: condition failed to evaluate within 45s: got 300 voters, want 298; details: map[1:r1:/{Min-System/NodeLiveness} [(n1,s1):1, (n3,s3):2, (n2,s2):3, (n5,s5):4, (n7,s7):5, (n4,s4):6, (n6,s6):7, next=8, gen=12] 2:r2:/System/NodeLiveness{-Max} [(n1,s1):1, (n6,s6):2, (n3,s3):3, (n2,s2):4, (n4,s4):5, next=6, gen=8] 3:r3:/System/{NodeLivenessMax-tsd} [(n1,s1):1, (n6,s6):2, (n7,s7):3, (n3,s3):4, (n5,s5):5, next=6, gen=8] 4:r4:/System{/tsd-tse} [(n4,s4):4, (n3,s3):2, (n7,s7):3, next=5, gen=8] 5:r5:/{Systemtse-Table/0} [(n6,s6):6, (n3,s3):2, (n7,s7):3, (n5,s5):4, (n2,s2):5, next=7, gen=12] 6:r6:/Table/{0-3} [(n3,s3):6, (n4,s4):2, (n6,s6):3, (n7,s7):4, (n2,s2):5, next=7, gen=12] 7:r7:/Table/{3-4} [(n1,s1):1, (n2,s2):2, (n6,s6):3, (n4,s4):4, (n7,s7):5, next=6, gen=8] 8:r8:/Table/{4-5} [(n1,s1):1, (n4,s4):2, (n7,s7):3, (n5,s5):4, (n3,s3):5, next=6, gen=8] 9:r9:/Table/{5-6} [(n6,s6):6, (n3,s3):2, (n5,s5):3, (n2,s2):4, (n4,s4):5, next=7, gen=12] 10:r10:/Table/{6-7} [(n1,s1):1, (n6,s6):2, (n5,s5):3, (n2,s2):4, (n4,s4):5, next=6, gen=8] 11:r11:/Table/{7-8} [(n1,s1):1, (n7,s7):2, (n6,s6):3, (n4,s4):4, (n5,s5):5, next=6, gen=8] 12:r12:/Table/{8-11} [(n1,s1):1, (n5,s5):2, (n7,s7):3, (n4,s4):4, (n6,s6):5, next=6, gen=8] 13:r13:/Table/1{1-2} [(n5,s5):6, (n2,s2):2, (n6,s6):3, (n7,s7):4, (n3,s3):5, next=7, gen=12] 14:r14:/Table/1{2-3} [(n6,s6):6, (n3,s3):2, (n7,s7):3, (n4,s4):4, (n5,s5):5, next=7, gen=12] 15:r15:/Table/1{3-4} [(n1,s1):1, (n7,s7):2, (n4,s4):3, (n5,s5):4, (n6,s6):5, next=6, gen=8] 16:r16:/Table/1{4-5} [(n1,s1):1, (n4,s4):2, (n6,s6):3, (n7,s7):4, (n2,s2):5, next=6, gen=8] 17:r17:/Table/1{5-6} [(n7,s7):6, (n4,s4):2, (n2,s2):3, (n6,s6):4, (n5,s5):5, next=7, gen=12] 18:r18:/Table/1{6-7} [(n1,s1):1, (n4,s4):2, (n5,s5):3, (n6,s6):4, (n3,s3):5, next=6, gen=8] 19:r19:/Table/1{7-8} [(n1,s1):1, (n7,s7):2, (n4,s4):3, (n5,s5):4, (n2,s2):5, next=6, gen=8] 20:r20:/Table/1{8-9} [(n1,s1):1, (n4,s4):2, (n2,s2):3, (n7,s7):4, (n6,s6):5, next=6, gen=8] 21:r21:/Table/{19-20} [(n6,s6):6, (n2,s2):2, (n4,s4):3, (n3,s3):4, (n7,s7):5, next=7, gen=12] 22:r22:/Table/2{0-1} [(n1,s1):1, (n5,s5):2, (n3,s3):3, (n4,s4):4, (n6,s6):5, next=6, gen=8] 23:r23:/Table/2{1-2} [(n1,s1):1, (n5,s5):2, (n3,s3):3, (n7,s7):4, (n4,s4):5, next=6, gen=8] 24:r24:/Table/2{2-3} [(n1,s1):1, (n6,s6):2, (n5,s5):3, (n7,s7):4, (n2,s2):5, next=6, gen=8] 25:r25:/Table/2{3-4} [(n1,s1):1, (n2,s2):2, (n3,s3):3, (n5,s5):4, (n4,s4):5, next=6, gen=8] 26:r26:/Table/2{4-5} [(n1,s1):1, (n2,s2):2, (n3,s3):3, (n6,s6):4, (n4,s4):5, next=6, gen=8] 27:r27:/Table/2{5-6} [(n4,s4):6, (n2,s2):2, (n5,s5):3, (n3,s3):4, (n7,s7):5, next=7, gen=12] 28:r28:/Table/2{6-7} [(n1,s1):1, (n6,s6):2, (n7,s7):3, (n4,s4):4, (n3,s3):5, next=6, gen=8] 29:r29:/Table/2{7-8} [(n1,s1):1, (n4,s4):2, (n2,s2):3, (n6,s6):4, (n3,s3):5, next=6, gen=8] 30:r30:/Table/2{8-9} [(n1,s1):1, (n5,s5):2, (n4,s4):3, (n2,s2):4, (n6,s6):5, next=6, gen=8] 31:r31:/{Table/29-NamespaceTable/30} [(n1,s1):1, (n5,s5):2, (n3,s3):3, (n4,s4):4, (n2,s2):5, next=6, gen=8] 32:r32:/NamespaceTable/{30-Max} [(n5,s5):6, (n3,s3):2, (n7,s7):3, (n2,s2):4, (n4,s4):5, next=7, gen=12] 33:r33:/{NamespaceTable/Max-Table/32} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n6,s6):4, (n3,s3):5, next=6, gen=8] 34:r34:/Table/3{2-3} [(n1,s1):1, (n7,s7):2, (n2,s2):3, (n3,s3):4, (n5,s5):5, next=6, gen=8] 35:r35:/Table/3{3-4} [(n1,s1):1, (n2,s2):2, (n3,s3):3, (n6,s6):4, (n5,s5):5, next=6, gen=8] 36:r36:/Table/3{4-5} [(n7,s7):6, (n4,s4):2, (n2,s2):3, (n5,s5):4, (n3,s3):5, next=7, gen=12] 37:r37:/Table/3{5-6} [(n1,s1):1, (n5,s5):2, (n4,s4):3, (n7,s7):4, (n3,s3):5, next=6, gen=8] 38:r38:/Table/3{6-7} [(n1,s1):1, (n6,s6):2, (n2,s2):3, (n4,s4):4, (n5,s5):5, next=6, gen=8] 39:r39:/Table/3{7-8} [(n1,s1):1, (n6,s6):2, (n5,s5):3, (n7,s7):4, (n3,s3):5, next=6, gen=8] 40:r40:/Table/3{8-9} [(n4,s4):6, (n5,s5):2, (n3,s3):3, (n2,s2):4, (n6,s6):5, next=7, gen=12] 41:r41:/Table/{39-40} [(n5,s5):6, (n2,s2):2, (n7,s7):3, (n6,s6):4, (n3,s3):5, next=7, gen=12] 42:r42:/Table/4{0-1} [(n1,s1):1, (n3,s3):2, (n6,s6):3, (n7,s7):4, (n2,s2):5, next=6, gen=8] 43:r43:/Table/4{1-2} [(n1,s1):1, (n6,s6):2, (n7,s7):3, (n2,s2):4, (n3,s3):5, next=6, gen=8] 44:r44:/Table/4{2-3} [(n3,s3):6, (n7,s7):2, (n5,s5):3, (n6,s6):4, (n4,s4):5, next=7, gen=12] 45:r45:/Table/4{3-4} [(n1,s1):1, (n2,s2):2, (n5,s5):3, (n3,s3):4, (n4,s4):5, next=6, gen=8] 46:r46:/Table/4{4-5} [(n1,s1):1, (n5,s5):2, (n6,s6):3, (n2,s2):4, (n7,s7):5, next=6, gen=8] 47:r47:/Table/4{5-6} [(n7,s7):6, (n4,s4):2, (n3,s3):3, (n5,s5):4, (n2,s2):5, next=7, gen=12] 48:r48:/Table/4{6-7} [(n1,s1):1, (n7,s7):2, (n3,s3):3, (n4,s4):4, (n2,s2):5, next=6, gen=8] 49:r49:/Table/4{7-8} [(n1,s1):1, (n5,s5):2, (n6,s6):3, (n3,s3):4, (n7,s7):5, next=6, gen=8] 50:r50:/Table/{48-50} [(n1,s1):1, (n6,s6):2, (n4,s4):3, (n5,s5):4, (n7,s7):5, next=6, gen=8] 51:r51:/Table/5{0-1} [(n1,s1):1, (n4,s4):2, (n2,s2):3, (n5,s5):4, (n6,s6):5, next=6, gen=8] 52:r52:/Table/5{1-2} [(n1,s1):1, (n7,s7):2, (n3,s3):3, (n6,s6):4, (n2,s2):5, next=6, gen=8] 53:r53:/Table/5{2-3} [(n1,s1):1, (n5,s5):2, (n6,s6):3, (n3,s3):4, (n7,s7):5, next=6, gen=8] 54:r54:/Table/5{3-4} [(n2,s2):6, (n4,s4):2, (n7,s7):3, (n6,s6):4, (n5,s5):5, next=7, gen=12] 55:r55:/Table/5{4-5} [(n1,s1):1, (n3,s3):2, (n6,s6):3, (n7,s7):4, (n2,s2):6, next=7, gen=12] 56:r56:/Table/5{5-6} [(n1,s1):1, (n6,s6):2, (n7,s7):3, (n2,s2):4, (n3,s3):5, next=6, gen=8] 57:r57:/Table/5{6-7} [(n1,s1):1, (n5,s5):2, (n2,s2):3, (n4,s4):4, (n7,s7):5, next=6, gen=8] 58:r58:/Table/5{7-8} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n6,s6):4, (n5,s5):5, next=6, gen=8] 59:r59:/Table/5{8-9} [(n1,s1):1, (n7,s7):2, (n4,s4):3, (n3,s3):4, (n6,s6):5, next=6, gen=8] 60:r60:/{Table/59-Max} [(n1,s1):1, (n3,s3):2, (n5,s5):3, (n2,s2):4, (n7,s7):5, next=6, gen=8]]
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestSystemZoneConfigs589320229
--- FAIL: TestSystemZoneConfigs (62.23s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSystemZoneConfigs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -25,7 +25,6 @@ import (
 	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
-	"github.com/cockroachdb/cockroach/pkg/config/zonepb"
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed/rangefeedcache"
@@ -35,7 +34,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/isolation"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverpb"
-	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvstorage"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptpb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptutil"
@@ -2684,118 +2682,6 @@ func TestChangeReplicasGeneration(t *testing.T) {
 	// +1 for transitioning out of joint config
 	// +1 for removing learner
 	assert.EqualValues(t, repl.Desc().Generation, oldGeneration+3, "\nold: %+v\nnew: %+v", oldDesc, newDesc)
-}
-
-func TestSystemZoneConfigs(t *testing.T) {
-	defer leaktest.AfterTest(t)()
-	defer log.Scope(t).Close(t)
-
-	skip.WithIssue(t, 98905)
-
-	// This test is relatively slow and resource intensive. When run under
-	// stressrace on a loaded machine (as in the nightly tests), sometimes the
-	// SucceedsSoon conditions below take longer than the allotted time (#25273).
-	skip.UnderRace(t)
-	skip.UnderShort(t)
-	skip.UnderStress(t)
-
-	ctx := context.Background()
-	tc := testcluster.StartTestCluster(t, 7, base.TestClusterArgs{
-		ServerArgs: base.TestServerArgs{
-			Knobs: base.TestingKnobs{
-				Store: &kvserver.StoreTestingKnobs{
-					// Disable LBS because when the scan is happening at the rate it's happening
-					// below, it's possible that one of the system ranges trigger a split.
-					DisableLoadBasedSplitting: true,
-				},
-			},
-			// This test was written for the gossip-backed SystemConfigSpan
-			// infrastructure.
-			DisableSpanConfigs: true,
-			// Scan like a bat out of hell to ensure replication and replica GC
-			// happen in a timely manner.
-			ScanInterval: 50 * time.Millisecond,
-		},
-	})
-	defer tc.Stopper().Stop(ctx)
-	log.Info(ctx, "TestSystemZoneConfig: test cluster started")
-
-	expectedSystemRanges, err := tc.Servers[0].ExpectedInitialRangeCount()
-	if err != nil {
-		t.Fatal(err)
-	}
-	expectedUserRanges := 1
-	expectedSystemRanges -= expectedUserRanges
-	systemNumReplicas := int(*zonepb.DefaultSystemZoneConfig().NumReplicas)
-	userNumReplicas := int(*zonepb.DefaultZoneConfig().NumReplicas)
-	expectedReplicas := expectedSystemRanges*systemNumReplicas + expectedUserRanges*userNumReplicas
-	log.Infof(ctx, "TestSystemZoneConfig: expecting %d system ranges and %d user ranges",
-		expectedSystemRanges, expectedUserRanges)
-	log.Infof(ctx, "TestSystemZoneConfig: expected (%dx%d) + (%dx%d) = %d replicas total",
-		expectedSystemRanges, systemNumReplicas, expectedUserRanges, userNumReplicas, expectedReplicas)
-
-	waitForReplicas := func() error {
-		replicas := make(map[roachpb.RangeID]roachpb.RangeDescriptor)
-		for _, s := range tc.Servers {
-			if err := kvstorage.IterateRangeDescriptorsFromDisk(ctx, s.Engines()[0], func(desc roachpb.RangeDescriptor) error {
-				if len(desc.Replicas().LearnerDescriptors()) > 0 {
-					return fmt.Errorf("descriptor contains learners: %v", desc)
-				}
-				if existing, ok := replicas[desc.RangeID]; ok && !existing.Equal(&desc) {
-					return fmt.Errorf("mismatch between\n%s\n%s", &existing, &desc)
-				}
-				replicas[desc.RangeID] = desc
-				return nil
-			}); err != nil {
-				return err
-			}
-		}
-		var totalReplicas int
-		for _, desc := range replicas {
-			totalReplicas += len(desc.Replicas().VoterDescriptors())
-		}
-		if totalReplicas != expectedReplicas {
-			return fmt.Errorf("got %d voters, want %d; details: %+v", totalReplicas, expectedReplicas, replicas)
-		}
-		return nil
-	}
-
-	// Wait until we're down to the expected number of replicas. This is
-	// effectively waiting on replica GC to kick in to destroy any replicas that
-	// got removed during rebalancing of the initial ranges, since the testcluster
-	// waits until nothing is underreplicated but not until all rebalancing has
-	// settled down.
-	testutils.SucceedsSoon(t, waitForReplicas)
-	log.Info(ctx, "TestSystemZoneConfig: initial replication succeeded")
-
-	// Update the meta zone config to have more replicas and expect the number
-	// of replicas to go up accordingly after running all replicas through the
-	// replicate queue.
-	sqlDB := sqlutils.MakeSQLRunner(tc.ServerConn(0))
-	sqlutils.SetZoneConfig(t, sqlDB, "RANGE meta", "num_replicas: 7")
-	expectedReplicas += 2
-	testutils.SucceedsSoon(t, waitForReplicas)
-	log.Info(ctx, "TestSystemZoneConfig: up-replication of meta ranges succeeded")
-
-	// Do the same thing, but down-replicating the timeseries range.
-	sqlutils.SetZoneConfig(t, sqlDB, "RANGE timeseries", "num_replicas: 1")
-	expectedReplicas -= 2
-	testutils.SucceedsSoon(t, waitForReplicas)
-	log.Info(ctx, "TestSystemZoneConfig: down-replication of timeseries ranges succeeded")
-
-	// Up-replicate the system.jobs table to demonstrate that it is configured
-	// independently from the system database.
-	sqlutils.SetZoneConfig(t, sqlDB, "TABLE system.jobs", "num_replicas: 7")
-	expectedReplicas += 2
-	testutils.SucceedsSoon(t, waitForReplicas)
-	log.Info(ctx, "TestSystemZoneConfig: up-replication of jobs table succeeded")
-
-	// Finally, verify the system ranges. Note that in a new cluster there are
-	// two system ranges, which we have to take into account here.
-	sqlutils.SetZoneConfig(t, sqlDB, "RANGE system", "num_replicas: 7")
-	expectedReplicas += 4
-	testutils.SucceedsSoon(t, waitForReplicas)
-	log.Info(ctx, "TestSystemZoneConfig: up-replication of system ranges succeeded")
 }
 
 func TestClearRange(t *testing.T) {

